fix for issue #15 BuiltIn.EventTimeStamp is not copied

martin-weber opened the pull request:
Fix for issue: If a property is bound to BuiltIn.EventTimeStamp its value isn't copied #15

In the case of $ts (BuiltIn.EventTimeStamp) the EventFiledName is not the property name but "$ts" (BuiltIn.EventTimeStamp)

maker-martin committed:
fix for: If a property is bound to BuiltIn.EventTimeStamp its value isn't copied #15

In the case of $ts (BuiltIn.EventTimeStamp) the EventFiledName is not the property name but "$ts" (BuiltIn.EventTimeStamp)

YohskDista commented on Chronological/QueryResults/Events/EventQueryResultToTypeMapper.cs:
```diff
@@ -62,6 +62,14 @@ internal IEnumerable<T> Map<T>(IEnumerable<EventResult> eventResults)
+                            if (propertyType.ToLower() == "datetime" &&
+                                (typeProperty.PropertyType == typeof(DateTime) || typeProperty.PropertyType == typeof(DateTime?)))
+                            {
+                                typeProperty.SetValue(instance, DateTime.Parse(value));
```
The parsing will throw an exception if `value` is null. I don't know if you make this on purpose but I think it can be dangerous

martin-weber replied:
I think this should not be an issue since the value for the $ts property should never be null
But you are right, I willll remove the "|| typeProperty.PropertyType == typeof(DateTime?)"

martin-weber replied:
see commit below

maker-martin committed:
remove possibility of nullable datetime to prevent errors

updated due to feedback on pullrequest
remove possibility of nullable datetime to prevent errors

colethecoder commented:
Thanks for this, it looks good to me. I'll merge now but it might not be until the weekend that I release the nuget package for it.
